src/StreamJsonRpc/JsonRpcEventSource.cs

```diff
@@ -0,0 +1,232 @@
+        [Event(InvokeMethodStartEvent, Level = EventLevel.Verbose)]
+        public void InvokeMethodStart(string message)
+        {
+            this.WriteEvent(InvokeMethodStartEvent, message);
+        }
+
+        /// <summary>
+        /// Invoke Method end.
+        /// </summary>
+        /// <param name="message">Request and method details.</param>
+        [Event(InvokeMethodStopEvent, Level = EventLevel.Verbose)]
+        public void InvokeMethodStop(string message)
+        {
+            this.WriteEvent(InvokeMethodStopEvent, message);
+        }
+
+        /// <summary>
+        /// Cancel Request start.
+        /// </summary>
+        /// <param name="message">Cancellation message.</param>
+        [Event(CancelRequestStartEvent, Level = EventLevel.Verbose)]
+        public void CancelRequestStart(string message)
+        {
+            this.WriteEvent(CancelRequestStartEvent, message);
+        }
+
+        /// <summary>
+        /// Cancel Request end.
+        /// </summary>
+        /// <param name="message">Cancellation message.</param>
+        [Event(CancelRequestStopEvent, Level = EventLevel.Verbose)]
+        public void CancelRequestStop(string message)
+        {
+            this.WriteEvent(CancelRequestStopEvent, message);
+        }
+
+        /// <summary>
+        /// Invoke Notification.
+        /// </summary>
+        /// <param name="message">Details of the Notification.</param>
+        [Event(InvokeNotificationStartEvent, Level = EventLevel.Verbose)]
+        public void InvokeNotificationStart(string message)
+        {
+            this.WriteEvent(InvokeNotificationStartEvent, message);
+        }
+
+        /// <summary>
+        /// Notification completed.
+        /// </summary>
+        /// <param name="message">Details of the Notification.</param>
+        [Event(InvokeNotificationStopEvent, Level = EventLevel.Verbose)]
+        public void InvokeNotificationStop(string message)
+        {
+            this.WriteEvent(InvokeNotificationStopEvent, message);
+        }
+
+        /// <summary>
+        /// Creates a string representation of arguments of max 1024 characters for logging.
+        /// </summary>
+        /// <param name="arguments">A single argument or an array of arguments.</param>
+        /// <returns>String representation of first argument only.</returns>
+        internal static string GetArgumentsString(object arguments)
+        {
+            if (arguments == null)
+            {
+                return string.Empty;
+            }
+
+            const int maxLength = 1024;
+
+            if (arguments is object[] args)
+            {
+                var stringBuilder = new StringBuilder();
+                if (args != null && args.Length > 0)
+                {
+                    for (int i = 0; i < args.Length; ++i)
+                    {
+                        stringBuilder.Append($"arg{i}: {args[i].ToString()}, ");
```

Comment:
You can drop the `.ToString()`. Formatting arguments *always* get that method called on them implicitly anyway.
Also, I wonder if rendering arguments like this is a good idea. Maybe objects don't override ToString(), and those that do may use quotes, colons, and commas anyway leading to a very confusing output here. 
As this is something of a tough problem and can lead to large output, do we *need* the include arguments? They wouldn't be needed as part of a perf investigation I think, and they're already available in the `TraceSource` log. Can we just remove ETW tracing of arguments to keep things simple for this first iteration?

Reply:
Without arguments sometimes its difficult to know which method will get called. 
For instance, in WinForms, Method is mostly: "SendRequest". Its the argument that tells which endpoint will get hit. E.g: 'control/GetVisible', 'object/GetPropertyValue' etc.

Reply:
Hmm... ok fair point. I'm curious why you factored it that way. You can't have more precise method names? I guess since you've already identified that it's important to you, that's fine.

Reply:
That is how the WinForms designer implementation is. I guess it was done to have a single end point on the client side and club related operations for each object type.

I can limit the length of arguments to say 200 characters instead of 1024.

Reply:
Truncated arguments to 128 chars. 
Also removed arguments from InvokeMethodStop and SendRequestStop events as they can be correlated with their Start counterparts using request Id.